frontend/lib/src/hooks/useSubmitFormViaEnterKey.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useCallback } from "react"
+
+import { WidgetStateManager } from "@streamlit/lib/src/WidgetStateManager"
+
+const isEnterKeyPressed = (
```

Comment:
**suggestion:** While we're here, let's update `frontend/lib/src/components/widgets/ChatInput/ChatInput.tsx` to also utilize this

Reply:
I had a quick look but it seems like `ChatInput.tsx` cannot be used inside forms as of today (see [here](https://github.com/streamlit/streamlit/blob/develop/frontend/lib/src/components/widgets/ChatInput/ChatInput.tsx#L245-L246)), so I don't think we can use the hook there.

Reply:
At the least, can we share this `isEnterKeyPressed` function?

Reply:
oh yeah, 100%! thanks a lot and also for looking at this with persistence, I have only had the hook in mind 😬

Reply:
Also, I have moved the `isEnterKeyPressed` function to a new `util/inputUtils.ts` file since its unrelated to forms.